contracts/game/src/lib.cairo

```diff
@@ -224,6 +225,45 @@ mod Game {
             _start_game(ref self, weapon, name, interface_camel);
         }
 
+        fn optimistic_start(ref self: ContractState, adventurer_id: felt252, block_hash: felt252) {
+            // assert the starting entropy has not been set
+            assert(
+                self._starting_entropy.read(adventurer_id) == 0, messages::STARTING_HASH_NOT_ZERO
+            );
+
+            self._starting_entropy.write(adventurer_id, block_hash);
+
+            // then attack the starting beast
+            self.attack(adventurer_id, false);
+        }
+
+
+        fn slay_invalid_adventurer(ref self: ContractState, adventurer_id: felt252) {
+            // get starting blockhash 
+            let optimistic_hash = self._starting_entropy.read(adventurer_id);
+
+            // must be optimistic start
+            assert(optimistic_hash != 0, messages::NO_OPTIMISTIC_START);
```

Comment:
See comment in `attack`. I believe the bypass would render `_starting_entropy` as 0 and therefore you cannot slay them for being invalid.